openerp/models.py

```diff
@@ -5740,7 +5740,8 @@ def onchange(self, values, field_name, field_onchange):
         """
         env = self.env
 
-        if field_name and field_name not in self._fields:
+        names = filter(None, field_name if isinstance(field_name, list) else [field_name])
```

Comment:
We have to keep current behavior.
If someone call `onchange()` without `field_name`, he expect the onchange to be called on all fields in `values`